syntax = "proto3";

message PersonMessage {
  int32 id = 1;
  string first_name = 2;
  string last_name = 3;
  string company_name = 4;
}

message PersonIdMessage {
  int32 id = 1;
}

message PersonMessageList {
  repeated PersonMessage persons = 1;
}

message Empty {

}

service PersonService {
    rpc Create(PersonMessage) returns (PersonMessage);
    rpc Get(Empty) returns (PersonMessageList);
    rpc GetPerson(PersonIdMessage) returns (PersonMessage);
}